delfin/drivers/hpe/hpe_3par/component_handler.py

```diff
@@ -0,0 +1,243 @@
+            total_cap = total_cap - internal_cap - spare_cap
+            # Free capacity (MiB) in the system
+            free_cap = int(
+                storage.get('freeCapacityMiB')) * consts.MiB_TO_Bytes
+            used_cap = total_cap - free_cap
+
+            # raw_capacity
+            raw_cap = int(
+                storage.get('totalCapacityMiB')) * consts.MiB_TO_Bytes
+            # subscribed_capacity
+            subscribed_cap = int(
+                storage.get('allocatedCapacityMiB')) * consts.MiB_TO_Bytes
+            s = {
+                'name': storage.get('name'),
+                'vendor': ComponentHandler.HPE3PAR_VENDOR,
+                'model': storage.get('model'),
+                'status': status,
+                'serial_number': storage.get('serialNumber'),
+                'firmware_version': storage.get('systemVersion'),
+                'location': storage.get('location'),
+                'total_capacity': total_cap,
+                'raw_capacity': raw_cap,  # raw_capacity
+                'subscribed_capacity': subscribed_cap,  # subscribed_capacity
+                'used_capacity': used_cap,
+                'free_capacity': free_cap
+            }
+        else:
+            # If no data is returned, it indicates that there
+            # may be a problem with the network or the device.
+            # Default return OFFLINE
+            s = {
+                'status': status
+            }
+        return s
+
+    def list_storage_pools(self, context):
+        try:
+            # Get list of Hpe3parStor pool details
+            pools = self.restclient.get_all_pools()
+            pool_list = []
+
+            if pools is not None:
+                members = pools.get('members')
+                for pool in members:
+                    # Get pool status  1=normal 2,3=abnormal 99=offline
+                    status = constants.StoragePoolStatus.OFFLINE
+                    if pool.get('state') == consts.STATUS_POOL_NORMAL:
+                        status = constants.StoragePoolStatus.NORMAL
+                    elif (pool.get('state') == consts.STATUS_POOL_DEGRADED
+                          or pool.get('state') == consts.STATUS_POOL_FAILED):
+                        status = constants.StoragePoolStatus.ABNORMAL
+
+                    # Get pool storage_type   default block
+                    pool_type = constants.StorageType.BLOCK
+                    usr_used = int(
+                        pool['UsrUsage']['usedMiB']) * consts.MiB_TO_Bytes
+                    sa_used = int(
+                        pool['SAUsage']['usedMiB']) * consts.MiB_TO_Bytes
+                    sd_used = int(
+                        pool['SDUsage']['usedMiB']) * consts.MiB_TO_Bytes
+                    usr_total = int(
+                        pool['UsrUsage']['totalMiB']) * consts.MiB_TO_Bytes
+                    sa_total = int(
+                        pool['SAUsage']['totalMiB']) * consts.MiB_TO_Bytes
```

Comment:
what is totalMiB and usedMiB? sa_used and sd_total is differentiated only based upon totalMiB and usedMiB?